axelrod/tests/unit/test_pickling.py

```diff
@@ -0,0 +1,335 @@
+class Flip(axl.Cooperator):
+    pass
+
+
+@st.FinalTransformer((D, D), name_prefix=None)
+class Final(axl.Cooperator):
+    pass
+
+
+@st.ForgiverTransformer(0.2, name_prefix=None)
+class Forgiver(axl.Cooperator):
+    pass
+
+
+@st.GrudgeTransformer(3, name_prefix=None)
+class Grudge(axl.Cooperator):
+    pass
+
+
+@st.InitialTransformer((C, D), name_prefix=None)
+class Initial(axl.Cooperator):
+    pass
+
+
+@st.JossAnnTransformer((0.2, 0.2), name_prefix=None)
+class JossAnn(axl.Cooperator):
+    pass
+
+
+strategies = [axl.Grudger, axl.TitForTat]
+probability = [.2, .3]
+
+
+@st.MixedTransformer(probability, strategies, name_prefix=None)
+class Mixed(axl.Cooperator):
+    pass
+
+
+@st.NiceTransformer(name_prefix=None)
+class Nice(axl.Cooperator):
+    pass
+
+
+@st.NoisyTransformer(0.2, name_prefix=None)
+class Noisy(axl.Cooperator):
+    pass
+
+
+@st.RetaliationTransformer(3, name_prefix=None)
+class Retaliation(axl.Cooperator):
+    pass
+
+
+@st.RetaliateUntilApologyTransformer(name_prefix=None)
+class RetaliateUntilApology(axl.Cooperator):
+    pass
+
+
+@st.TrackHistoryTransformer(name_prefix=None)
+class TrackHistory(axl.Cooperator):
+    pass
+
+@st.IdentityTransformer()
+class Identity(axl.Cooperator):
+    pass
+
+transformed_no_prefix = [Apology, DeadlockBreaking, Flip, Final, Forgiver,
+                         Grudge, Initial, JossAnn, Mixed, Nice, Noisy,
+                         Retaliation, RetaliateUntilApology, TrackHistory, Dual,
+                         Identity]
+
+transformer_instances = [
+    st.ApologyTransformer([D], [C]),
+    st.DeadlockBreakingTransformer(),
+    st.DualTransformer(),
+    st.FlipTransformer(),
+    st.FinalTransformer((D, D)),
+    st.ForgiverTransformer(0.2),
+    st.GrudgeTransformer(3),
+    st.InitialTransformer((C, D)),
+    st.JossAnnTransformer((0.2, 0.6)),
+    st.MixedTransformer(probability, strategies),
+    st.NiceTransformer(),
+    st.NoisyTransformer(0.2),
+    st.RetaliationTransformer(3),
+    st.RetaliateUntilApologyTransformer(),
+    st.TrackHistoryTransformer(),
+    st.IdentityTransformer()
+]
+
+
+class TestPickle(unittest.TestCase):
+
+    def assert_orignal_equals_pickled(self, player, turns=10):
+        self.assert_original_plays_same_as_pickled(player, turns)
+        self.assert_mutated_instance_same_as_pickled(player)
+
+    def assert_original_plays_same_as_pickled(self, player, turns=10):
+        copy = pickle.loads(pickle.dumps(player))
+        opponent_1 = axl.CyclerCCCDCD()
+        opponent_2 = axl.CyclerCCCDCD()
+        axl.seed(0)
+        match_1 = axl.Match((player, opponent_1), turns=turns)
+        result_1 = match_1.play()
+
+        axl.seed(0)
+        match_2 = axl.Match((copy, opponent_2), turns=turns)
+        result_2 = match_2.play()
+
+        self.assertEqual(result_1, result_2)
+
+    def assert_mutated_instance_same_as_pickled(self, player):
+        player.reset()
+        turns = 5
+        opponent = axl.Alternator()
```

Comment:
adding a few seconds to total test time is not an issue, correct?  since the play mutates the original player, perhaps i can just absorb this into the previous test.